db/migrations/44_replace_items_items_always_visible_and_access_restricted_with_partial_access_propagation.sql

```diff
@@ -0,0 +1,269 @@
+-- +migrate Up
+ALTER TABLE `items_items` ADD COLUMN `partial_access_propagation` enum('None','AsGrayed','AsPartial') NOT NULL DEFAULT 'None'
+    COMMENT 'Specifies how to propagate partial access to the child item' AFTER `access_restricted`;
+ALTER TABLE `history_items_items` ADD COLUMN `partial_access_propagation`
+    enum('None','AsGrayed','AsPartial') NOT NULL DEFAULT 'None' AFTER `access_restricted`;
+
+UPDATE `items_items` SET `partial_access_propagation` = IF(`access_restricted`, IF(`always_visible`, 'AsGrayed', 'None'), 'AsPartial');
+UPDATE `history_items_items` SET `partial_access_propagation` = IF(`access_restricted`, IF(`always_visible`, 'AsGrayed', 'None'), 'AsPartial');
+
+ALTER TABLE `items_items` DROP COLUMN `access_restricted`, DROP COLUMN `always_visible`;
+ALTER TABLE `history_items_items` DROP COLUMN `access_restricted`, DROP COLUMN `always_visible`;
+
+DROP TRIGGER IF EXISTS `after_insert_items_items`;
+-- +migrate StatementBegin
+CREATE DEFINER=`algorea`@`%` TRIGGER `after_insert_items_items` AFTER INSERT ON `items_items` FOR EACH ROW BEGIN
+    INSERT INTO `history_items_items` (`id`,`version`,`item_parent_id`,`item_child_id`,`child_order`,`category`,
+                                       `partial_access_propagation`,`difficulty`)
+        VALUES (NEW.`id`,@curVersion,NEW.`item_parent_id`,NEW.`item_child_id`,NEW.`child_order`,NEW.`category`,
+                NEW.`partial_access_propagation`,NEW.`difficulty`);
+    INSERT IGNORE INTO `groups_items_propagate`
+        SELECT `id`, 'children' as `propagate_access` FROM `groups_items`
+            WHERE `groups_items`.`item_id` = NEW.`item_parent_id`
+            ON DUPLICATE KEY UPDATE propagate_access='children';
+END
+-- +migrate StatementEnd
+DROP TRIGGER IF EXISTS `before_update_items_items`;
+-- +migrate StatementBegin
+CREATE DEFINER=`algorea`@`%` TRIGGER `before_update_items_items` BEFORE UPDATE ON `items_items` FOR EACH ROW BEGIN
```

Comment:
Waw I didn't expect such a big migration. 

Do we want to keep these triggers (ignoring history stuffs)? Maybe that's better if we consider we'll add FK with possible "delete cascade" which may be hard to address in services.

Reply:
These triggers are mostly related to items access rights propagation and items ancestors creation.